src/frontend/src/utils/condition.rs

```diff
@@ -415,35 +415,61 @@ impl Condition {
                     || matches!(func_type, ExprType::GreaterThan))
             {
                 let mut pk_struct = vec![];
+                let mut order_type = None;
                 let mut all_added = true;
                 let mut iter = row_left_inputs.iter().zip_eq_fast(right_iter);
-                for i in 0..table_desc.order_column_indices().len() {
+                for column_order in &table_desc.pk {
                     if let Some((left_expr, right_expr)) = iter.next() {
-                        if left_expr.as_input_ref().unwrap().index != i {
+                        if left_expr.as_input_ref().unwrap().index != column_order.column_index {
                             all_added = false;
                             break;
                         }
+                        match order_type {
+                            Some(o) => {
+                                if o != column_order.order_type {
+                                    all_added = false;
+                                    break;
+                                }
+                            }
+                            None => order_type = Some(column_order.order_type),
+                        }
```

Comment:
I think the check here will pass if the PKs are all ordered desc:
```
create materialized view mv1 as select * from t1 order by v1 desc, v2 desc, v3 desc;
```

However, in this case, I think the optimization is not applicable.

Reply:
We just need to make sure that the order type of all the pk's is the same, in the row_seq_scan_exec will generate the appropriate scan according to the order type of the first pk to scan the data.
https://github.com/risingwavelabs/risingwave/blob/fe84f69e1611dad37f0ea97ff9dcdcbbaa972b63/src/batch/executors/src/executor/row_seq_scan.rs#L438

Reply:
Got it. That means when all pks are ordered desc, with the following scan range:
```
ExprType::GreaterThan => {
    (Bound::Included(pk_struct), Bound::Unbounded)
}
```

the storage scan will be done by scanning from `Bound::Unbounded` to `Bound::Included(pk_struct)`